org.eclipse.xtext.common.types.ui/src/org/eclipse/xtext/common/types/ui/notification/DeltaConverter.java

```diff
@@ -187,21 +189,38 @@ protected void convertChangedCompilationUnit(IJavaElementDelta delta, List<IReso
 		
 		ICompilationUnit compilationUnit = (ICompilationUnit) delta.getElement();
 		try {
-			for (IType type : compilationUnit.getTypes()) {
+			doForeachCompilationType(compilationUnit, type -> {
 				if (isDerived(type))
```

Comment:
Here we do have a changed semantic: formally, we would return from the method if the typ was a derived type. With the change, only the lambda is left and the processing would continue for the remaining types in the compilation unit. Also the convertRemovedTypes below would kick in. Could you please sort this out?

Reply:
I think it's ok now